src/ethereum/EnigmaContractProductionWriterAPI.js

```diff
@@ -34,24 +35,36 @@ class EnigmaContractProductionWriterAPI extends EnigmaContractWriterAPI {
         from: res.transactionOptions.from,
         to: this._enigmaContractAddress,
         gas: res.transactionOptions.gas,
-        // this encodes the ABI of the method and the arguments
+        // this encodes the ABI of the method and th
         data: this._enigmaContract.methods.register(utils.add0x(signerAddress), utils.add0x(report), utils.add0x(signature)).encodeABI()
       };
 
-      this._web3.eth.accounts.signTransaction(tx, this._privateKey)
-        .then((signedTx)=> {
-          this._web3.eth.sendSignedTransaction(signedTx.rawTransaction)
-            .on(ETHEREUM_ERROR_EVENT, (error, receipt) => {
-              reject(error);
-            })
-            .on(ETHEREUM_RECEIPT_EVENT, (receipt) => {
-              resolve(null);
-            })
-        })
-        .catch((error) =>{
+      try {
+        const signedTx = await this._web3.eth.accounts.signTransaction(tx, this._privateKey)
+        const signedTransaction = this._web3.eth.sendSignedTransaction(signedTx.rawTransaction).on(ETHEREUM_ERROR_EVENT, (error, receipt) => {
           reject(error);
-        });
-    });
+        })
+          .on(ETHEREUM_RECEIPT_EVENT, async receipt => {
+            if (!this.minimumConfirmations) {
```

Comment:
`ETHEREUM_CONFIRMATION_EVENT` starts from `1`, this is why it's necessary to also use `ETHEREUM_RECEIPT_EVENT` here, in case it's `0` or not a number.  
I'll fix `!this.minimumConfirmations` to be `this.minimumConfirmations == 0 || !Number.isInteger(minimumConfirmations)`.